File: 백준/Silver/5525. IOIOI/IOIOI.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = +input.shift();
const str = "I" + "OI".repeat(N);
const len = str.length;
const M = +input.shift();
let start = 0;
let cnt = 0;
while (start <= M - len) {
  let flag = false;
  for (let i = 0; i < len; i++) {
    if (input[0][i + start] !== str[i]) {
      flag = true;
      break;
    }
  }
  if (!flag) cnt++;
  start++;
}
console.log(cnt);

File: 백준/Silver/5525. IOIOI/IOIOI.py
import sys
input = sys.stdin.readline

N = int(input())
M = int(input())
S = input().strip()

pattern = "I" + "OI" * N
p_len = len(pattern)

cnt = 0
start = S.find("I")     

while start != -1 and start <= M - p_len:
    if S[start:start + p_len] == pattern:
        cnt += 1
    start += 1

print(cnt)